src/repositories/request-repository.ts

```diff
@@ -5,13 +5,18 @@ import { BaseRepository } from 'typeorm-transactional-cls-hooked';
 import { Request, RequestUpdateFields } from "../models/request";
 import { RequestStatus } from "../models/request-status";
 import { logEvent } from '../logger';
-import { config } from "node-config-ts";
-import { singleton } from "tsyringe";
+import { Config } from 'node-config-ts';
+import { inject, singleton } from 'tsyringe';
 
 @singleton()
 @EntityRepository(Request)
 export default class RequestRepository extends BaseRepository<Request> {
 
+  constructor(
+    @inject('config') private config?: Config) {
```

Comment:
I am curious, do you know @stbrody why the argument has to be marked as optional here?

Reply:
I was just copying what I saw done elsewhere.  I suspect it's for unit tests that might not choose to register all possible components?